### ISSUE TITLE
Bug: [no-unsafe-assignment] Differentiate a types-error any from a "true" any

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8231]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=DYUwLgBAbghsCuIBcEB28C2AjEAnA3AFCGwIgQC8EAzgA4D29A1gJ75A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolp9kySgHMmAW0Qt0URNGgdokcGAC%2BITUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
let value: number;

value = spooky;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unsafe-assignment": "error"
  }
}
```

### Expected Result

Users should be told that the type is an "error" variant of `any`. It's confusing to have the same report show up for both a "real" `any` and an error in type checking.

### Actual Result

The same error as if we'd added a `declare const spooky: any;`:

```plaintext
Unsafe assignment of an `any` value.
```

### Additional Info

Although the type's `typeToString()` returns `"any"`, its intrinsic name is `"error"`. See the _Types_ panel of the linked playground.

Error types in type checking are often a symptom of misconfiguration around TypeScript and/or typescript-eslint. [Example StackOverflow question around an `any` in a lint rule](https://stackoverflow.com/questions/77659106/overcome-eslints-redundant-type-constituents-for-disposable-interface).

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages%2Ftype-utils%2Fsrc%2Fpredicates.ts#L88-L96

We already have some handling for this - though not every callsite uses that util..

**Comment by JoshuaKGoldberg** [comment]
Since #8304 is `1 approval`, retargeting this issue to just be fore `no-unsafe-assignment`. I'll file separate issues for the other rules that it impacts.

**Comment by JoshuaKGoldberg** [comment]
Filed:

* #9214
* #9215
* #9216
* #9217
* #9218
